rcl/include/rcl/macros.h

```diff
@@ -20,12 +20,8 @@ extern "C"
 {
 #endif
 
-#ifndef _WIN32
 /// Ignored return values of functions with this macro will emit a warning.
-# define RCL_WARN_UNUSED __attribute__((warn_unused_result))
-#else
-# define RCL_WARN_UNUSED _Check_return_
-#endif
+#define RCL_WARN_UNUSED RCUTILS_WARN_UNUSED
```

Comment:
@dirk-thomas same here about unrelated changes. Again, I'd be perfectly fine with this changes in a separate commit so they don't get collapsed into the same commit about unused return values.

Reply:
I think this should stay together since this change makes sure that the signature is actually the same.

Reply:
BTW https://github.com/ros2/rcl/pull/653 does both.